plugin-build/plugin/src/main/java/io/sentry/kotlin/multiplatform/gradle/plugin/SentryPlugin.kt

```diff
@@ -0,0 +1,231 @@
+    kmpExtension.targets.forEach { target ->
+        if (unsupportedTargets.any { unsupported ->
+                target.name.contains(unsupported)
+            }
+        ) {
+            throw GradleException(
+                "Unsupported target: ${target.name}. " +
+                    "Cannot auto install in commonMain. " +
+                    "Please create an intermediate sourceSet with targets that the SDK supports and add the dependency manually.",
+            )
+        }
+    }
+
+    val commonMain =
+        kmpExtension.sourceSets.find {
+            it.name.contains("common")
+        }
+
+    val sentryVersion = commonMainAutoInstallExtension.sentryKmpVersion.get()
+    commonMain?.dependencies {
+        api("io.sentry:sentry-kotlin-multiplatform:$sentryVersion")
+    }
+}
+
+internal fun Project.installSentryForCocoapods(cocoapodsAutoInstallExtension: CocoapodsAutoInstallExtension) {
+    val kmpExtension = extensions.findByName(KOTLIN_EXTENSION_NAME)
+    if (kmpExtension !is KotlinMultiplatformExtension) {
+        return
+    }
+
+    (kmpExtension as ExtensionAware).extensions.configure(CocoapodsExtension::class.java) { cocoapods ->
+        val podName = "Sentry"
+        val sentryPod = cocoapods.pods.findByName(podName)
+        if (sentryPod == null) {
+            cocoapods.pod(podName) {
+                version = cocoapodsAutoInstallExtension.sentryCocoaVersion.get()
+                linkOnly = true
+                extraOpts += listOf("-compiler-option", "-fmodules")
+            }
+        }
+    }
+}
+
+internal fun Project.configureLinkingOptions(linkerExtension: LinkerExtension) {
+    val kmpExtension = extensions.findByName(KOTLIN_EXTENSION_NAME)
+    if (kmpExtension !is KotlinMultiplatformExtension) {
+        // todo: log, not multiplatform found
+        return
+    }
+
+    val customXcodeprojPath = linkerExtension.xcodeprojPath.orNull
+    val derivedDataPath = findDerivedDataPath(customXcodeprojPath)
+
+    kmpExtension.appleTargets().all { target ->
+        val frameworkArchitecture = target.toSentryFrameworkArchitecture()
+        if (frameworkArchitecture == null) {
+            // todo: log, unsupported architecture
+            return@all
+        }
+
+        val dynamicFrameworkPath =
+            "$derivedDataPath/SourcePackages/artifacts/sentry-cocoa/Sentry-Dynamic/Sentry-Dynamic.xcframework/$frameworkArchitecture"
+        val staticFrameworkPath =
+            "$derivedDataPath/SourcePackages/artifacts/sentry-cocoa/Sentry/Sentry.xcframework/$frameworkArchitecture"
+
+        target.binaries.all binaries@{ binary ->
+            val path =
+                when {
+                    File(dynamicFrameworkPath).exists() -> dynamicFrameworkPath
+                    File(staticFrameworkPath).exists() -> {
+                        // todo: log, dynamic framework not found, try using static framework
+                        staticFrameworkPath
+                    }
```

Comment:
`h`: iirc, if we link the static framework to a dynamic kmp library, it will crash at runtime right? can/should we also check if the binary `isStatic` in this case? Basically have supported combinations of framework-klib